### PR TITLE
Update basedocs.jl for $

### PR DESCRIPTION
**Pull request by Delta456**
Task for Google Code In 2019

Add docstring of '$' for interpolation at the REPL. Solves https://github.com/JuliaLang/julia/issues/33666

### CODE REVIEW

**Comment by Delta456** [comment]
Need help with falling builds

**Comment by Delta456** [comment]
Could anyone help me what's wrong?

**Comment by jaideep-seth** [comment]
Take at solving #33666

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -447,6 +447,22 @@ julia> A'
 """
 kw"'"
 
+"""
+    \$
+
+The string interpolation operator, see [string](@ref string-interpolation and [expression](@ref man-expression-interpolation).
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: I think we would need more details here :sweat_smile: 

You can have a look at docstrings of `=` or other operators.

**Comment by jaideep-seth** [comment]
Please see the basedocs.jl of this merged pull request (https://github.com/JuliaLang/julia/pull/33344/files)

**Comment by ViralBShah** [comment]
A big welcome to all the Code-In contributors.

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -447,6 +447,22 @@ julia> A'
...
+
+# Examples
+```jldoctest
+julia> name = Joe
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: This is incorrect. It should be `name = "Joe"`

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -447,6 +447,22 @@ julia> A'
...
+Joe
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: And here too: `"Joe"`

**Review by abhishalya — CHANGES_REQUESTED** [review]

**Comment by abhishalya** [comment]
Also, this needs a rebase.

File: base/docs/basedocs.jl (hunk 3 above)
Comment (Delta456) [reply base/docs/basedocs.jl]: damn

File: base/docs/basedocs.jl [hunk 4]
```diff
@@ -448,20 +448,20 @@ julia> A'
 kw"'"
 
 """
-    \$
+    '$'
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: Why did you change this, it was correct before.

File: base/docs/basedocs.jl [hunk 5]
```diff
@@ -448,20 +448,20 @@ julia> A'
...
+julia> name = "Joe"
+"Joe"
 
 julia> "My name is $name."
-My name is Joe.
+"My name is Joe."
 ```
 """
-kw"$"
+kw'$'
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: Again, why? Change it to `kw"\$"`.

**Review by abhishalya — CHANGES_REQUESTED** [review]

**Comment by jaideep-seth** [comment]
Hi @Delta456 , this is a sample PR that I've made, please get your basedocs.jl correct. Create an issue for your failing builds and u will accept the task!

**Comment by Delta456** [comment]
@jaideep-seth I think the compiler should also allow `"$"` instead of giving an error for string interpolation.

File: base/docs/basedocs.jl [hunk 6]
```diff
@@ -447,6 +447,23 @@ julia> A'
...
+
+The string interpolation operator, see [string](@ref string-interpolation)
+and [expression](@ref man-expression-interpolation).
+
+# Examples
+```jldoctest
+julia> name = "Joe"
+"Joe"
+
+julia> "My name is $name."
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: We completely forgot, there's a `$` here too. We need to escape this as well.

**Review by abhishalya — CHANGES_REQUESTED** [review]

File: base/docs/basedocs.jl (hunk 6 above)
Comment (abhishalya) [reply base/docs/basedocs.jl]: You can easily check if you've done correctly or not by using the `println` function.

```julia
julia> some_str = """My name is $name."""
```
This will produce an error:
```
UndefVarError: name not defined
```

File: base/docs/basedocs.jl [hunk 7]
```diff
@@ -447,6 +447,23 @@ julia> A'
 """
 kw"'"
 
+"""
+    "\$"
```
Comment (abhishalya) [inline base/docs/basedocs.jl]: No `" "` here please. This should be just `\$`

**Review by abhishalya — CHANGES_REQUESTED** [review]

File: base/docs/basedocs.jl (hunk 7 above)
Comment (Delta456) [reply base/docs/basedocs.jl]: Why

File: base/docs/basedocs.jl (hunk 6 above)
Comment (Delta456) [reply base/docs/basedocs.jl]: This is the whole example.

```julia
# Examples
julia> name = "Joe"
"Joe"

julia> "My name is $name."
"My name is Joe."
```
Output

```console
julia> name = "Joe"
"Joe"

julia> """My name is $name"""
"My name is Joe"
```

Comment (abhishalya) [reply base/docs/basedocs.jl]: See the point is, you are writing this inside triple quotes `"""` which translates to string. Here Julia needs to process these strings and hence in that process the `$` should be escaped. Else, it will look for `name` which you don't have. I hope you understand.

Nevertheless, can you make the changes I suggested and push, if it all goes well the build should be green.

Comment (Delta456) [reply base/docs/basedocs.jl]: I think the compiler allow using `$` and I now I get it. Thanks

**Comment by jaideep-seth** [comment]
Hello @abhishalya & @Delta456 , please take up some of these task instances - (https://codein.withgoogle.com/tasks/5659574762209280/?sp-page=5)

**Comment by abhishalya** [comment]
@jaideep-seth Thanks, but I think I might just leave that for new people since we've already done some Beginner tasks.

**Review by StefanKarpinski — APPROVED** [review]
This looks good to me. Thanks for the doc improvement. Anyone familiar with documenter stuff know if this is the right way to attach docs to the `$` symbol? Maybe @fredrikekre?

**Review by abhishalya — CHANGES_REQUESTED** [review]
This needs a rebase!

Also, it would be much better to have the commits squashed into one.

**Comment by StefanKarpinski** [comment]
It's easy to squash the commits into a single one through the GitHub interface, so don't worry about that :)
